### PR TITLE
Update project README

### PR DESCRIPTION
**Pull request by adrianchiris**
Update README providing better overview
of the project, prereqisites and how to use
the operator.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,14 +2,35 @@
...
+
+<small><i><a href='http://ecotrust-canada.github.io/markdown-toc/'>Table of contents generated with markdown-toc</a></i></small>
+
+# Nvidia Mellanox Network Operator
 Nvidia Mellanox Network Operator leverages [Kubernetes CRDs](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/)
 and [Operator SDK](https://github.com/operator-framework/operator-sdk) to manage Networking related Components in order to enable Fast networking, 
 RDMA and GPUDirect for workloads in a Kubernetes cluster.
 
-## Prerequisites
+The Goal of Network Operator is to manage _all_ networking related bits a pieces to enable execution of
```
Comment (moshe010) [inline README.md:26]: related bits a pieces?

File: README.md [hunk 2]
```diff
@@ -42,18 +63,90 @@ The Operator Acts on the following CRDs:
...
+          }
+        ]
+      }
+```
+
+Can be found at: `example/deploy/crds/mellanox.com_v1alpha1_nicclusterpolicy_cr.yaml`
+
+#### NICClusterPolicy status:
+NICClusterPolicy `status` field reflects the current state of the system.
+it Contains a per sub-state/stage and a global state `status` that indicates if the cluster has transitioned to the desired
```
Comment (moshe010) [inline README.md:113]: s/it Contains/It contains

Comment (moshe010) [reply README.md:113]: All Line 113 and 115 it too long and complicate sentence

File: README.md [hunk 3]
```diff
@@ -42,18 +63,90 @@ The Operator Acts on the following CRDs:
...
+  State:    ready
+```
+
+>__NOTE__: An `ignore` State indicates that the sub-state/stage was not defined in the custom resource
+> this it is ignored.
 
 ## System Requirements
-As more driver containers are built the operator will be able to support additional platforms.
-for Phase 1 Ubuntu20.04 is planned to be supported.
+* RDMA capable hardware: Mellanox ConnectX-5 NIC.
```
Comment (moshe010) [inline README.md:135]: We can say it will work on Mellanox ConnectX-4 and above and just tested in ConnectX-5

File: README.md [hunk 4]
```diff
@@ -62,3 +155,21 @@ An initialization script loads the modules when the container is run (in privile
 making them available to the kernel.
 
 While this approach may seem odd. It provides a way to deliver drivers to immutable systems.
+
+[Mellanox OFED and NV Peer Memory driver container](https://github.com/Mellanox/ofed-docker)
+
+## Future work
```
Comment (moshe010) [inline README.md:161]: let remove the future work for now

File: README.md (hunk 1 above)
Comment (adrianchiris) [reply README.md:26]: bits _and_ pieces ?
or do you prefer the more formal : "related components"

File: README.md (hunk 3 above)
Comment (adrianchiris) [reply README.md:135]: sure

File: README.md (hunk 4 above)
Comment (adrianchiris) [reply README.md:161]: sure

File: README.md (hunk 3 above)
Comment (adrianchiris) [reply README.md:135]: i dont think we need to say here on what we tested

File: README.md (hunk 2 above)
Comment (adrianchiris) [reply README.md]: This has been re-worded, let me know if that ok

**Pull request merged**